replace_substring_in_filenames.sh

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env bash
+#
+# Name: replace_substring_in_filenames.sh
+#
+# Brief: Command-line Bash script to remove repeated substring in multiple 
+# LibreOffice .odt files in 1 directory - so filenames are shorter & 
+# easier-to-read. Takes in 3 parameters at command line, e.g.,
+# directory path:       /Users/kimlew/Sites/bash_projects/test_rename_files
+# string to replace:    'Inkscape Essent Train-'
+# replacement_string:   nothing or ''
+#
+# REMEMBER: If command-line arguments have spaces, put in single-quotes!
+# Note: Script processes only a single directory. 
+#
+# Author: Kim Lew
+
+# Check if arguments are provided. If no arguments given, display user prompts 
+# of what to enter.
+if [ $# -eq 0 ]; then
+  echo "Type the directory path for filenames that need changes: "
+  read directory_path
+  
+  echo "Type the string to replace: "
+  read string_to_replace
+
+  echo "Type the replacement string: "
+  read replacement_string
+else
+  if [ ! "$1" ]; then
+    echo "Enter the directory path as the 1st parameter." 
+    exit 1
+  fi
+  if [ ! "$2" ]; then
+    echo "Enter the string to replace as the 2nd parameter." 
+    exit 1
+  fi
+  if [ ! "$3" ]; then
+    replacement_string=''
+  fi
```

Comment:
You should be able to remove this third `if` block without breaking your script.

If I'm not mistaken the test `[ ! "$3" ]` will pass only if  `"$3"` evaluates to the empty string. This happens when the user gives no third argument at all, or gives an explicit empty string using quotes (`""` or `''`). Either way, the assignment

```
  replacement_string=$3
```

below on line 42 will give `replacement_string` an empty value. In fact, that assignment proves that this `if` block isn't necessary.

Reply:
@kevin Please re-review.
- As you pointed out,  I removed the unneeded 3rd command-line argument check & re-tested
- Also, to correct for another issue I found, to keep spaces entered with user prompt responses, I added before the ```read``` commands of the user prompts:
``` IFS='' ```

Reply:
Looks good